### PR TITLE
holo-auth-client: init

### PR DESCRIPTION
**Pull request by samrose**

### CODE REVIEW

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+def main():
+    #temp creation of json
+    data = {
+            "email":"test@test.com"
+    }
+    with open('/var/lib/data.json', 'w') as outfile:
+        json.dump(data, outfile)
+    #end temp
+    path = '/var/lib/data.json'
+    datafile = open(path, 'r')
```
[the remark this answers is not in the document]
Comment (yegortimoshenko) [reply overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]: Also, this line has trailing whitespace (`autopep8` will fix this).

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    with open('/var/lib/data.json', 'w') as outfile:
+        json.dump(data, outfile)
+    #end temp
+    path = '/var/lib/data.json'
+    datafile = open(path, 'r')    
+    loadedjson = json.load(datafile)
+    get_zt = subprocess.run(["zerotier-cli", "-j", "info"], capture_output=True)
+    #print(get_zt)
+    ztjson = json.loads(get_zt.stdout.decode('utf-8'))
+    print(ztjson)
```
Comment (yegortimoshenko) [inline overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]:
```suggestion
```

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    }
+    with open('/var/lib/data.json', 'w') as outfile:
+        json.dump(data, outfile)
+    #end temp
+    path = '/var/lib/data.json'
+    datafile = open(path, 'r')    
+    loadedjson = json.load(datafile)
+    get_zt = subprocess.run(["zerotier-cli", "-j", "info"], capture_output=True)
+    #print(get_zt)
+    ztjson = json.loads(get_zt.stdout.decode('utf-8'))
```
Comment (yegortimoshenko) [inline overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]: You actually don't need to decode it, `json.loads` will work on bytestrings as-is:

```suggestion
    ztjson = json.loads(get_zt.stdout))
```

File: overlays/holo-nixpkgs/holo-whitelist/default.nix [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ python3 ];
+
+  buildCommand = ''
+    makeWrapper ${python3}/bin/python3 $out/bin/holo-whitelist \
+      --add-flags ${./holo-whitelist.py} 
+  '';
+
+  meta.platforms = platforms.linux;
+}
```
Comment (yegortimoshenko) [inline overlays/holo-nixpkgs/holo-whitelist/default.nix]: :+1:

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    print(ztjson)
+    payload = {
+            'zt_address' : ztjson['address'],
+            'email' : loadedjson['email']
+    }
+    url = "https://postman-echo.com/post"
```
Comment (yegortimoshenko) [inline overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]: This should probably be a constant.

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import subprocess
+
+def main():
+    #temp creation of json
+    data = {
+            "email":"test@test.com"
+    }
+    with open('/var/lib/data.json', 'w') as outfile:
+        json.dump(data, outfile)
+    #end temp
```
[the remark this answers is not in the document]
Comment (samrose) [reply overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]: Done, it is removed

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (samrose) [reply overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]: autotpep8'd

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    path = '/var/lib/data.json'
+    datafile = open(path, 'r')    
+    loadedjson = json.load(datafile)
```
[the remark this answers is not in the document]
Comment (samrose) [reply overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]: It does now, thanks for catching that

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    get_zt = subprocess.run(["zerotier-cli", "-j", "info"], capture_output=True)
+    #print(get_zt)
```
[the remark this answers is not in the document]
Comment (samrose) [reply overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]: done

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py (hunk 2 above)
Comment (samrose) [reply overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]: done

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py (hunk 5 above)
Comment (samrose) [reply overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]: Now it is

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import json
+import requests
+import subprocess
+
```
[the remark this answers is not in the document]
Comment (samrose) [reply overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]: Done

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py (hunk 3 above)
Comment (samrose) [reply overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]: removed

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+def main():
+    #temp creation of json
+    data = {
+            "email":"test@test.com"
+    }
+    with open('/var/lib/data.json', 'w') as outfile:
+        json.dump(data, outfile)
+    #end temp
+    path = '/var/lib/data.json'
```
[the remark this answers is not in the document]
Comment (samrose) [reply overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]: now it is

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    datafile = open(path, 'r')    
+    loadedjson = json.load(datafile)
+    get_zt = subprocess.run(["zerotier-cli", "-j", "info"], capture_output=True)
```
[the remark this answers is not in the document]
Comment (samrose) [reply overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]: done

File: overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py [hunk 12]
```diff
@@ -0,0 +1,29 @@
+import json
+import requests
+import subprocess
+
+def main():
```
[the remark this answers is not in the document]
Comment (samrose) [reply overlays/holo-nixpkgs/holo-whitelist/holo-whitelist.py]: done

**Comment by samrose** [comment]
@yegortimoshenko do you approve merging this?

**Comment by samrose** [comment]
(fixed merge conflict)

**Comment by samrose** [comment]
@yegortimoshenko on this one, I believe I have addressed all of your requested changes fwiw

**Comment by yegortimoshenko** [comment]
@samrose: Could you review c529d47b67db3d65562bf5cc02f742741678d4b1...d2cc6c46b3ec6963c3432d91035141e6f160f712?

File: profiles/holoportos/demo/default.nix [hunk 13]
```diff
@@ -32,7 +32,13 @@ in
...
   imports = [ ../. ];
 
-  environment.systemPackages = [ pkgs.holo-init pkgs.holo-keygen ];
+  environment.systemPackages = [
+    pkgs.holo-config
+    pkgs.holo-init
+    pkgs.holo-keygen
+    pkgs.holofuel-demo-configure
+    pkgs.holo-whitelist
+  ];
```
Comment (yegortimoshenko) [inline profiles/holoportos/demo/default.nix]: Looks like bad merge.

Comment (samrose) [reply profiles/holoportos/demo/default.nix]: @yegortimoshenko changes look good

**Pull request merged**